bofire/strategies/doe/design.py

```diff
@@ -572,7 +199,7 @@ def find_local_max_ipopt(
     if _ipopt_options[b"print_level"] > 12:  # type: ignore
```

Comment:
In my opinion this could be removed. The objective criterion can be viewed via the IPOPT output and any other criterion can also be calculated with little effort by creating another `Objective`. What do you think? :) --> then there is also no need anymore to implement get_model_matrix() for `SpaceFilling`.
Also the metrics(), g_optimality(), a_optimality(), d_optimality() functions in doe utils could be deleted then since they are not used anywhere else.